### ISSUE TITLE
Subscription to typing notifications

### ISSUE THREAD

**Issue by j4hangir** [tinode/chat#599]
I looked into API and couldn't find the proper way to receive `kp` events without subscribing to topics first.

This doesn't make a lot of sense, is there a API for this that I'm missing, or should this be considered a new feature?

P.s. 
As far as I understand it's also not advisable to subscribe to many topics all at the same time.

**Comment by or-else** [comment]
> receive kp events without subscribing to topics first

That's by design.

> This doesn't make a lot of sense

Why? Say, you have 1000 topics, all sending `kp`s. Why is it beneficial to get them all?

The `kp`s are explicitly filtered out here: 
https://github.com/tinode/chat/blob/next/server/topic.go#L927
If you really want to have them, just add them to the check and you will get them all on 'me' topic.

**Comment by j4hangir** [comment]
This is a typical feature ,when you're looking at your contacts, if any of them is writing you won't be notified _unless_ you're subscribed to them already. This isn't how most messengers work, you'll see if they're typing right there.

Thanks for the aforementioned like, I'll look into it 👍🏻

**Comment by j4hangir** [comment]
That line certainly isn't responsible for that event, also another question, I'm not even getting `read` and `recv` without subscribing to the topic! Is that also intentional?

**Comment by or-else** [comment]
Why don't you spend a bit of time learning the code?

**Comment by j4hangir** [comment]
I've already modified the server and tested it with the aforementioned line, it didn't work. 

I've also got the original instance of Tinode running on my server, and testing back-and-forth I don't see any `read` and `recv` being exchanged before opening a contact, and looking at the design layout:

![image](https://user-images.githubusercontent.com/7769125/104081131-f22ce800-522c-11eb-8b0d-57c9d9ae1fb2.png)


It's not implemented there neither, in other words, we won't know if a contact has read or received the message *before* clicking on their name.

In conclusion I think I've spent more than _a little time_, now, is this all intentional or am I missing something?

**Comment by j4hangir** [comment]
The design layout I'm working on is similar to WhatsApp, etc. and the same contacts above are shown as this:

![image](https://user-images.githubusercontent.com/7769125/104081282-e4c42d80-522d-11eb-9378-e784e605bd23.png)

Aside from the `kb` issue, I can't find a proper way to be informed of `read` and `recv` without subscribing to all existing topics (which is probably a bad idea).

**Comment by or-else** [comment]
https://github.com/tinode/chat/pull/595

**Comment by or-else** [comment]
This is done.
